From befdbeeaaf93764e7169c5e159b9fb5c82dc67bb Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:05 -0400
Subject: [powerpc] prom: Stop scanning dev-tree for fdump early

Message-id: <1410545655-205645-497-git-send-email-dzickus@redhat.com>
Patchwork-id: 94413
O-Subject: [RHEL7 PATCH 496/626] powerpc/prom: Stop scanning dev-tree for fdump early
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit a7d0431774cf4b9970cf49da5b5cdd2f1aa71b25
Author: Gavin Shan <gwshan@linux.vnet.ibm.com>
Date:   Thu Apr 24 18:00:31 2014 +1000

    powerpc/prom: Stop scanning dev-tree for fdump early

    Function early_init_dt_scan_fw_dump() is called to scan the device
    tree for fdump properties under node "rtas". Any one of them is
    invalid, we can stop scanning the device tree early by returning
    "1". It would save a bit time during boot.

    Signed-off-by: Gavin Shan <gwshan@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/fadump.c b/arch/powerpc/kernel/fadump.c
index 2230fd0..0266774 100644
--- a/arch/powerpc/kernel/fadump.c
+++ b/arch/powerpc/kernel/fadump.c
@@ -69,7 +69,7 @@ int __init early_init_dt_scan_fw_dump(unsigned long node,
   */
  token = of_get_flat_dt_prop(node, "ibm,configure-kernel-dump", NULL);
  if (!token)
-  return 0;
+  return 1;
 
  fw_dump.fadump_supported = 1;
  fw_dump.ibm_configure_kernel_dump = *token;
@@ -92,7 +92,7 @@ int __init early_init_dt_scan_fw_dump(unsigned long node,
      &size);
 
  if (!sections)
-  return 0;
+  return 1;
 
  num_sections = size / (3 * sizeof(u32));
 
@@ -110,6 +110,7 @@ int __init early_init_dt_scan_fw_dump(unsigned long node,
    break;
   }
  }
+
  return 1;
 }
 
-- 
1.7.1